test/metabase/models/entity_id_test.clj

```diff
@@ -0,0 +1,68 @@
+(ns metabase.models.entity-id-test
+  "To support serialization, all exported entities should have either an external name (eg. a database path) or a
+  generated NanoID in a column called entity_id. There's a property :entity_id to automatically populate that field.
+
+  This file makes it impossible to forget to add entity_id to new entities. It tests that every entity is either
+  explicitly excluded, or has the :entity_id property."
+  (:require
+    [clojure.test :refer [deftest is]]
+    [metabase.models]
+    [toucan.models :as models :refer [IModel]]))
+
+(def entities-external-name
+  "Entities with external names, so they don't need a generated entity_id."
+  #{;; Databases have external names based on their URLs; tables are nested under databases; fields under tables.
+    metabase.models.database.DatabaseInstance
+    metabase.models.table.TableInstance
+    metabase.models.field.FieldInstance
+    ;; Settings have human-selected unique names.
+    metabase.models.setting.SettingInstance
+    ;; Dependencies are serialized but no extra ID is necessary since they're just many-to-many links between entities
+    ;; with unique IDs, and don't need unique IDs of their own.
+    metabase.models.dependency.DependencyInstance})
```

Comment:
nit: this is not sorted

Reply:
I didn't intend to sort this list, but rather to capture the semantic order databases -> tables -> fields.
Should I just sort it instead?

Reply:
ah, then I think we should leave that unsorted.